pkg/agent/flowexporter/exporter/exporter_perf_test.go

```diff
@@ -145,6 +146,39 @@ func BenchmarkExportDenyConns(b *testing.B) {
 
 }
 
+func NewFlowExporterForTest(o *flowexporter.FlowExporterOptions) (*FlowExporter, error) {
+	// Initialize IPFIX registry
+	registry := ipfix.NewIPFIXRegistry()
+	registry.LoadRegistry()
+
+	// Prepare input args for IPFIX exporting process.
+	nodeName := "test-node"
+	expInput := prepareExporterInputArgs(o.FlowCollectorAddr, o.FlowCollectorProto, nodeName)
+
+	v4Enabled := true
+	v6Enabled := false
+
+	denyConnStore := connections.NewDenyConnectionStore(nil, nil, o)
+	conntrackConnStore := connections.NewConntrackConnectionStore(nil, v4Enabled, v6Enabled, nil, nil, nil, o)
+
+	return &FlowExporter{
+		conntrackConnStore:     conntrackConnStore,
+		denyConnStore:          denyConnStore,
+		registry:               registry,
+		v4Enabled:              v4Enabled,
+		v6Enabled:              v6Enabled,
+		exporterInput:          expInput,
+		ipfixSet:               ipfixentities.NewSet(false),
+		k8sClient:              nil,
+		nodeRouteController:    nil,
+		isNetworkPolicyOnly:    false,
+		nodeName:               nodeName,
+		conntrackPriorityQueue: conntrackConnStore.GetPriorityQueue(),
+		denyPriorityQueue:      denyConnStore.GetPriorityQueue(),
+		expiredConns:           make([]flowexporter.Connection, 0, maxConnsToExport*2),
+	}, nil
```

Comment:
nit: since you are making a dedicated initialization function for testing purpose and it should never fail as there should be no external dependency in unit test, there is no point to have the error return value. You ignored it anyway in the caller.

Reply:
thanks, updated